numpy/_pytesttester.py

```diff
@@ -137,13 +137,20 @@ def __call__(self, label='fast', verbose=1, extra_argv=None,
         # offset verbosity. The "-q" cancels a "-v".
         pytest_args += ["-q"]
 
-        # Filter out distutils cpu warnings (could be localized to
-        # distutils tests). ASV has problems with top level import,
-        # so fetch module for suppression here.
         with warnings.catch_warnings():
             warnings.simplefilter("always")
+            # Filter out distutils cpu warnings (could be localized to
+            # distutils tests). ASV has problems with top level import,
+            # so fetch module for suppression here.
             from numpy.distutils import cpuinfo
 
+            # Ignore the warning from importing the array_api submodule. This
+            # warning is done on import, so it would break pytest collection,
+            # but importing it early here prevents the warning from being
+            # issued when it imported again.
+            warnings.simplefilter("ignore")
```

Comment:
I think you can omit this, `catch_warnings` should take care of it.

Reply:
It is needed because of the `simplefiler('always')` on line 141 above. Also catch_warnings doesn't disable warnings, it just resets the warnings filter at the end of the context. It looks like I can use `catch_warnings(record=True)` to make the test happy (and also make the warning not show/error when the tests are run). Unless this also doesn't work for the installed tests. I'll find out once I push I guess. 

Not sure what the point of that test is. Presumably someone wanted to make sure warnings weren't accidentally turned off anywhere.

Reply:
Delete the test if you like (maybe in a different PR).  The reason for it, was that Python was buggy: a warning, once ignored, was often always ignored.

Reply:
I'm not clear on the intent of the original. Was it to show a all warnings from `cpu_info` and suppress them thereafter since they were already raised?

Reply:
I'm not sure about the distutils stuff, but it seems like it might be similar to this, where the error happens at import time. The reason it won't be raised again is not because the warning is already raised. It's that Python only executes an import once, and caches it in sys.modules after that. But it's also why the guard has to go here. Otherwise, the warning gets issued at pytest collection time when it tries to import the module for the first time, and warnings also happen to be turned into errors during collection, so without this, the whole tests fail immediately. And the pytest collection seems to ignore the normal pytest warnings filters, hence this code here. I'm no pytest expert, so there might be a better way to do it.